### PR TITLE
NC | CI | Add Versioning tests to nc coretest

### PR DESCRIPTION
**Pull request by romayalon**
### Explain the changes
1. Add test_bucketspace_versioning.js and test_nsfs_versioning.js to nc_index.js.
2. Add NC_CORETEST changes for test_bucketspace_versioning.js.

### Issues: Fixed #xxx / Gap #xxx
1. 

### Testing Instructions:
1. Run `sudo NC_CORETEST=true node ./node_modules/mocha/bin/mocha src/test/unit_tests/test_bucketspace_versioning.js`
3. `sudo ./node_modules/mocha/bin/mocha src/test/unit_tests/test_nsfs_versioning.js`

- [ ] Doc added/updated
- [x] Tests added

### CODE REVIEW

**Review by shirady — COMMENTED** [review]
I added a few comments

**Comment by shirady** [comment]
@romayalon In the PR description in the "Testing Instructions" part I think you need to change the command:
`sudo ./node_modules/mocha/bin/mocha src/test/unit_tests/test_nsfs_versioning.js`
to 
`sudo NC_CORETEST=true node ./node_modules/mocha/bin/mocha src/test/unit_tests/test_nsfs_versioning.js`
(missing env variable of `NC_CORETEST` and the `node` command).

**Comment by romayalon** [comment]
> @romayalon In the PR description in the "Testing Instructions" part I think you need to change the command: `sudo ./node_modules/mocha/bin/mocha src/test/unit_tests/test_nsfs_versioning.js` to `sudo NC_CORETEST=true node ./node_modules/mocha/bin/mocha src/test/unit_tests/test_nsfs_versioning.js` (missing env variable of `NC_CORETEST` and the `node` command).

@shirady test_nsfs_versioning.js does not require NC_CORETEST=true, this tests works the same on all environments because it's calling directly to NamespaceFS, it's not doing rpc calls, db interaction etc.

File: docs/NooBaaNonContainerized/CI&Tests.md [hunk 1]
```diff
@@ -113,6 +113,18 @@ The following is a list of `NC jest tests` files -
...
+* `new_buckets_path` -
+    1. On NC environment - new_buckets_path is full absolute path.
+    2. On Containerized environment - new_buckets_path is the directory that will be concatenated to the namespace resource path.
+
+* `Multiple bucket exports the same directory` - 
+    1. On NC environment - Allowed.
+    2. On Containerized environment - Not Allowed.
+
+* `system owner / admin`
+    1. On NC environment - Do not exist.
```
[the remark this answers is not in the document]
Comment (romayalon) [reply docs/NooBaaNonContainerized/CI&Tests.md:130]: In all the versions the user does not know we have system owner on the bucket config files

**Comment by guymguym** [comment]
@romayalon does this PR enable the versioning test-cases of ceph s3-tests on nsfs?

**Review by guymguym — APPROVED** [review]

File: docs/NooBaaNonContainerized/CI&Tests.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (shirady) [reply docs/NooBaaNonContainerized/CI&Tests.md:130]: The assumption that the user is not exposed to the output of the noobaa-cli status output on the bucket and it is done by some system admin, right?

**Review by shirady — APPROVED** [review]
LGTM

**Comment by shirady** [comment]
> @shirady test_nsfs_versioning.js does not require NC_CORETEST=true, this test works the same on all environments because it's calling directly to NamespaceFS, it's not doing rpc calls, db interaction etc.

@romayalon I thought we needed it as long as in the test there is the condition `if (is_nc_coretest)`, because some tests behave differently.

**Comment by romayalon** [comment]
@shirady That's a correct assumption, but there is no if(is_nc_coretest) condition in test_nsfs_versioning.js...

**Pull request merged**